client/components/social-buttons/github.tsx

```diff
@@ -52,11 +56,19 @@ const GitHubLoginButton = ( {
 		}
 	}, [ socialServiceResponse ] );
 
+	const isDisabled = isFormDisabled || disabledState;
+
 	const handleClick = ( e: MouseEvent< HTMLButtonElement > ) => {
 		errorRef.current = e.currentTarget;
-	};
+		e.preventDefault();
 
-	const isDisabled = Boolean( disabledState || isFormDisabled || errorState );
+		if ( onClick ) {
+			onClick();
+		}
+
+		const clientId = config( 'github_oauth_client_id' );
+		window.location.href = `https://github.com/login/oauth/authorize?client_id=${ clientId }&redirect_uri=${ redirectUri }`;
```

Comment:
Do we have a preference where the GH sign-up form should open? At the moment it is the same tab (similarly to Continue with Apple flow).

Reply:
Our preference is same window, unless that doesn't work out. So all good!